packages/mcfunction/test/parser/entry.spec.ts

```diff
@@ -0,0 +1,35 @@
+import {
+	mockProjectData,
+	showWhitespaceGlyph,
+	testParser,
+} from '@spyglassmc/core/test-out/utils.js'
+import { describe, it } from 'mocha'
+import snapshot from 'snap-shot-it'
+import { initialize } from '../../../java-edition/lib/mcfunction/index.js'
+import { entry } from '../../lib/parser/index.js'
+import { tree } from './utils.js'
+
+describe('mcfunction parser entry()', () => {
+	const cases: { content: string }[] = [
+		{ content: '' },
+		{ content: 'say hi' },
+		{ content: 'say hi\nsay hi' },
+		{ content: '# this is a comment' },
+		{ content: '$ this is a macro command' },
+		{ content: '# this is a comment\nsay hi\n$this is a macro \n' },
+		{ content: 'execute if true if true run say hi' },
+		{ content: '# this is a comment \\ \n still a comment' },
+		{ content: '$ this is a macro \\ \n this is still a macro' },
+		{ content: 'sa\\  \n  y \\ \n hi' },
+		{ content: 'say trailing \\\n data' },
+		{ content: 'say \\\n hi \n # comment start \\\n end \n say hi' },
+	]
+	for (const { content } of cases) {
+		it(`Parse "${showWhitespaceGlyph(content)}"`, () => {
+			const release = '1.99'
+			initialize(mockProjectData(), tree, release)
```

Comment:
I'm not sure if it's a good idea to use the global state `initialize` function in a test. But I see that fixing that would require refactoring how the mcfunction parser works in a way where you pass it a tree instead of just the tree name.

Reply:
> fixing that would require refactoring how the mcfunction parser works in a way where you pass it a tree instead of just the tree name

yeah exactly. i wanted to minimize setup lines for these tests, this felt reasonable to me. but i'm open to alternatives

Reply:
found a good alternative. it was much easier than i thought it'd be

Reply:
also fixes the build!

Reply:
Much nicer!